### PR TITLE
make mitmproxy ignore non Genshin hosts

### PR DESCRIPTION
**Pull request by KawaiiNahida**
Since we set up a global proxy for the system, mitmproxy will try to modify the tls certificates of all hosts, which will prevent other applications on the system from running properly, so we need to add this command line parameter to make mitmproxy skip those non-Genshin related domains

### CODE REVIEW

File: start.cmd [hunk 1]
```diff
@@ -75,7 +75,7 @@ for /f "tokens=2*" %%a in ('reg query "HKCU\Software\Microsoft\Windows\CurrentVe
 @rem TODO: External proxy when ORIG_PROXY_ENABLE == 0x1
 echo set ws = createobject("wscript.shell") > "%temp%\proxy.vbs"
 echo ws.currentdirectory = "%MITMDUMP_PATH%" >> "%temp%\proxy.vbs"
-echo ws.run "cmd /c mitmdump.exe -s "^&chr(34)^&"%PROXY_SCRIPT_NAME%"^&chr(34)^&" -k",0 >> "%temp%\proxy.vbs"
+echo ws.run "cmd /c mitmdump.exe -s "^&chr(34)^&"%PROXY_SCRIPT_NAME%"^&chr(34)^&" -k --allow-hosts \".*\.yuanshen\.com|.*\.mihoyo\.com|.*\.hoyoverse\.com\"",0 >> "%temp%\proxy.vbs"
```
Comment (MlgmXyysd) [inline start.cmd]: You can't use escape symbols (`\"`) in vbs to handle quote nesting, use `"^&chr(34)^&"` instead.

**Review by MlgmXyysd — CHANGES_REQUESTED** [review]

**Comment by KawaiiNahida** [comment]
I'm sorry that I'm not good at writing VBS, I don't know if I have it right now

**Review by MlgmXyysd — CHANGES_REQUESTED** [review]
CR: Syntax error

**Comment by KawaiiNahida** [comment]
抱歉打扰了，现在应该是好了
图片

**Review by MlgmXyysd — CHANGES_REQUESTED** [review]
CR: Syntax error

**Comment by KawaiiNahida** [comment]
图片
图片
运行是正常的，生成vbs也是正确的，请问一下还有哪里有问题
It runs fine, and the generated vbs are correct, please tell me what else is wrong

**Review by MlgmXyysd — APPROVED** [review]
Sorry, I read it wrong, no issue here.

Merge proposal: Squash &amp; Merge

**Comment by ghost** [comment]
Please update your pull request to target the development branch,

**Review by KingRainbow44 — APPROVED** [review]

**Pull request merged**